### DISCUSSION TITLE
Host that runs the server is not served by the DHCP server

### DISCUSSION THREAD

**Discussion by mpeter50** [TechnitiumSoftware/DnsServer#817] [Q&A] [answered]
Hi!

I have recently set up the DNS server for a network connected to this Debian system (system A), along with the DHCP server functionality. The network does not have any other DHCP server.

For this network interface of the system I did not set up a regular static IP in `/etc/network/interfaces`, but instead I have set it up to use DHCP, and configured my DHCP client - dhclient - to use a fallback IP address until the DHCP server can respond.
I did so because the DHCP server does not start serving a network scope until the system has an address on it, but besides setting up other hosts (e.g. system B), I was also hoping that I could manage the network setup of system A through the DHCP server.

Then I have created a scope for this network, and have set up system A's dhclient DHCP client to use a fallback IP address on this network interface for use until it receives a response from the DHCP server, at which point it will be filled in on the details, like the network domain (option 15), NTP and DNS servers and such.

An other system (system B) on this network is served fine by the DHCP server, I see its lease on the web interface, and the dhclient logs in the system journal also show that 3/3 times it was served immediately, within the second.

But it seems that for system A something is wrong. It sends a DHCPREQUEST multiple times a minute, but it never receives a response:
```
dec 30 23:48:27 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:48:45 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:48:55 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:49:11 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:49:18 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:49:31 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:49:48 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:49:56 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
dec 30 23:50:14 infrastructurevm dhclient[16770]: DHCPREQUEST for 10.100.1.1 on ens19 to 255.255.255.255 port 67
```

dhclient here is set up to request 10.100.1.1 from the server (which runs on this same host), which serves the network of the ens19 interface. Here is its corresponding configuration:
```
lease {
    interface "ens19";
    fixed-address 10.100.1.1;
    option subnet-mask 255.255.255.0;
    renew 0 1970/01/01 00:00:00;
    rebind 0 1970/01/01 00:00:00;
    expire never;
}


timeout 1;
retry 30;
```

dhclient also has an other lease block for an other network that is served by an other DHCP server, and that one works fine.

Technitium-DNS did not log anything related to DHCP in the meantime, other than updating records for system B, and saving the scope file.
The confgiuration of the scope looks like this:

![image](https://github.com/TechnitiumSoftware/DnsServer/assets/83356418/f6d51401-eb45-471e-a538-0d3187524224)

The scope has been enabled, while the default one disabled:

![image](https://github.com/TechnitiumSoftware/DnsServer/assets/83356418/3123c753-c1d1-482d-822e-b23e4a00dc35)

Ping checks are not enabled, the requested address is not excluded from the scope, I have even made a reservation for it by MAC address, but it did not help.
I was thinking if maybe the DHCP server would ignore it because its the first host address in the network, but I'm not sure.

What could be the problem here?

My goal is to have a centralized configuration for the network services (NTP, DNS, network domain) to be used by hosts on this network, which would set up the DHCP server's system too.

**Comment by ShreyasZare** [comment]
Thanks for the details. The DHCP server itself cannot be a DHCP client. The server must have a static IP address for the DHCP server to work. So, the setup you are trying to do wont work. You have to configure all these network options manually on the server.

**Reply by mpeter50** [reply]
Thanks for the answer, thats unfortunate. Earlier I was looking at the network traffic with tcpdump, and it looked like as if the broadcast packets were just being sent but not received, do I understand it correctly that this is kind of a limitation of the operating system?

For configuring it manually, do you think it is a workable solution to set a "static" DHCP lease in `/etc/dhcp/dhclient.conf`?
Under this "static" lease I mean a `lease` block with an IP and all the network options defined there, with `renew` `rebind` and `expire` set to `never`, and of course also to set a reservation/exclusion in the DHCP scope of the server so that it never leases that IP to another host.

**Reply by ShreyasZare** [reply, marked as answer]
Its not unfortunate, its a requirement to run DHCP server. Imagine that you reboot the server and now it needs to renew lease but it itself is DHCP server and unable to listen since it does not have lease to the IP address. Also, the broadcast packets that are sent are not read by the same host as you observed.

So, you should just configure static IP address manually on the server. You also need to either configure the scope to exclude the IP ranges to avoid them being leased or specify a different start address for the scope.

Note that the DHCP server will not start if it detects that the IP address was assigned by DHCP. So configuring "static" lease will have issue with that.
